src/Features/Core/Portable/CustomMessageHandler/ICustomMessageHandlerService.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host;
+
+namespace Microsoft.CodeAnalysis.CustomMessageHandler;
+
+public interface ICustomMessageHandlerService : IWorkspaceService
+{
+    ValueTask<string> HandleCustomMessageAsync(
+        Solution solution,
+        string assemblyFolderPath,
```

Comment:
I'd just take `assemblyPath`. Makes it simpler to call and removes ambiguity of whether or not `assemblyFileName` includes extension.

Reply:
The reason for separating the folder path from the file name is that we want to load different assemblies from the same folder in the same assembly load context. So the `UnloadCustomMessageHandlersAsync` takes the folder path as parameter since it must unload the load context that contains all assemblies from that folder. So I think it's more coherent to have the folder path as a separate parameter in `HandleCustomMessageAsync` too.